projects/packages/publicize/src/class-rest-controller.php

```diff
@@ -453,7 +460,9 @@ public function update_publicize_connection( $request ) {
 			$body,
 			'wpcom'
 		);
-		return rest_ensure_response( $this->make_proper_response( $response ) );
+
+		global $publicize;
+		return rest_ensure_response( $publicize->get_connections_for_user( (int) $connection_id ) );
```

Comment:
Should we have some way of forcing this to be uncached? I think what's happening is after the connection is updated an XMLRPC request will get made so the cache is busted and the connections data will be updated, we'll then get the new connection data here.

But, we could get into some race conditions there. If the XMLRPC request is delayed, then we'll be reading the old connection data at this point.

Reply:
I have added a param here to force refresh. Is that better? When the page loads initially we can fire a call without the force refresh param. Behind the scenes, should we force refresh after the page loads?